### PR TITLE
Created gRPC client module

### PR DESCRIPTION
**Pull request by TylerStAmour**
Used some generics to hopefully make things simple when submitting tags which does limit backwards compatibility to Go 1.18+. I also only added a test for CompleteTest() since it ends up testing SubmitTag anyway, let me know if that's fine.

### CODE REVIEW

File: results/client/client.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+type ResultsClient struct {
+	conn   *grpc.ClientConn
+	client proto.TagTunnelClient
+}
+
+func (r *ResultsClient) Open(ctx context.Context) error {
+	conn, err := grpc.DialContext(ctx, addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return err
```
Comment (samparent97) [inline results/client/client.go:22]: for all error handling, user `errors.Wrap`. Put the name of the function that was called in lower case. I'll put up a couple examples.
```suggestion
		return errors.Wrap(err, "dial context")
```

File: results/client/client.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	}
+	r.conn = conn
+	r.client = proto.NewTagTunnelClient(conn)
+	return nil
+}
+
+func (r *ResultsClient) SubmitTag(ctx context.Context, tag string, value any) (bool, error) {
+	request, err := createRequest(tag, value)
+	if err != nil {
+		return false, err
```
Comment (samparent97) [inline results/client/client.go:32]:
```suggestion
		return false, errors.Wrap(err, "create request")
```

File: results/client/client.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import (
+	"context"
+	"github.com/google/uuid"
+	proto "github.com/macformula/hil/results/client/generated"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const addr = "localhost:31763"
```
Comment (samparent97) [inline results/client/client.go:12]: Add a constructor instead: 

```go
func NewResultsClient(ip, port string) *ResultsClient
```

File: results/client/client.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+func (r *ResultsClient) SubmitTag(ctx context.Context, tag string, value any) (bool, error) {
+	request, err := createRequest(tag, value)
+	if err != nil {
+		return false, err
+	}
+
+	reply, err := r.client.SubmitTag(ctx, request)
+	if err != nil {
+		return reply.Success, err
+	}
```
Comment (samparent97) [inline results/client/client.go:38]: Here you should use `reply.Success` to check the error message inside the reply. It's basically there to tell if the tag submitted actually exists. Keep the error check above as well though. 
```go
	if !reply.Success {
	    return false, errors.New(reply.Error)
	}
```

File: results/client/client.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	return reply.Success, nil
```
Comment (samparent97) [inline results/client/client.go:39]: The bool that this returns should be `reply.IsPassing`

File: results/client/client.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	case int32:
+		request.Data = &proto.SubmitTagRequest_ValueInt{ValueInt: data.(int32)}
+	case float32:
+		request.Data = &proto.SubmitTagRequest_ValueFloat{ValueFloat: data.(float32)}
+	case string:
+		request.Data = &proto.SubmitTagRequest_ValueStr{ValueStr: data.(string)}
+	case bool:
+		request.Data = &proto.SubmitTagRequest_ValueBool{ValueBool: data.(bool)}
+	default:
+		return nil, errors.New("unsupported data type for tag submission")
```
Comment (samparent97) [inline results/client/client.go:70]:
```suggestion
		return nil, errors.Errorf("unsupported data type for tag submission (%T)", data)
```

**Comment by TylerStAmour** [comment]
Should be good now

**Pull request merged**